typing/printtyp.ml

```diff
@@ -1903,36 +1900,75 @@ type printing_status =
       type error.
   *)
 
-let printing_status  = function
-  | Trace.(Diff { got=t1, t1'; expected=t2, t2'}) ->
-      if  is_constr_row ~allow_ident:true t1'
-       || is_constr_row ~allow_ident:true t2'
-      then Discard
-      else if same_path t1 t1' && same_path t2 t2' then Optional_refinement
-      else Keep
+let diff_printing_status { Errortrace.got=t1, t1'; expected=t2, t2'} =
+  if  is_constr_row ~allow_ident:true t1'
+   || is_constr_row ~allow_ident:true t2'
+  then Discard
+  else if same_path t1 t1' && same_path t2 t2' then Optional_refinement
+  else Keep
+
+type 'variety trace_format = {
```

Comment:
I agree with this proposed simplification.

Reply:
I went with an approach where it is implemented as a single-field record and left abstract; this can then easily be changed in the future.

Reply:
I updated this again to used the proposed approach, which dose seem nicer.  I also no longer expose this type, because we instead expose three separate `report_{unify,equality,moregen}_error` functions.  This can still be changed in the future if things become more appealing to do that way.